physical/mongo.go

```diff
@@ -0,0 +1,1121 @@
+package physical
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/mgutz/logxi/v1"
+
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// this part was derived from and adjusted for our needs from:
```

Comment:
Sure, I will try to do that. I was not doing it on purpose so far because I didn't want to change too many files/things for a first PR.